src/components/text-editor/__internal__/toolbar/toolbar.component.tsx

```diff
@@ -10,67 +15,116 @@ import Events from "../../../../__internal__/utils/helpers/events";
 import Icon from "../../../icon";
 import Tooltip from "../../../tooltip";
 import useLocale from "../../../../hooks/__internal__/useLocale";
+import {
+  BOLD,
+  ITALIC,
+  UNORDERED_LIST,
+  ORDERED_LIST,
+  InlineStyleType,
+  BlockType,
+} from "../../types";
 
-const BOLD = "BOLD";
-const ITALIC = "ITALIC";
-const UNORDERED_LIST = "unordered-list-item";
-const ORDERED_LIST = "ordered-list-item";
+export interface ToolbarProps {
+  /** Used to override the active status of the inline controls */
+  activeControls: Record<InlineStyleType | BlockType, boolean>;
+  /** Flag to trigger control focusing */
+  canFocus?: boolean;
+  /** Callback to handle setting the inline styles */
+  setInlineStyle: (
+    ev:
+      | React.KeyboardEvent<HTMLButtonElement>
+      | React.MouseEvent<HTMLButtonElement>,
+    inlineType: InlineStyleType
+  ) => void;
+  /** Callback to handle setting the block styles */
+  setBlockStyle: (
+    ev:
+      | React.KeyboardEvent<HTMLButtonElement>
+      | React.MouseEvent<HTMLButtonElement>,
+    blockType: BlockType
+  ) => void;
+  /** Additional elements to be rendered in the Toolbar, e.g. Save and Cancel Button */
+  toolbarElements?: React.ReactNode;
+}
 
 const Toolbar = ({
   activeControls,
   canFocus,
   toolbarElements,
   setBlockStyle,
   setInlineStyle,
-}) => {
+}: ToolbarProps) => {
   const { textEditor } = useLocale();
   const { tooltipMessages, ariaLabels } = textEditor;
-  const controlRefs = [useRef(), useRef(), useRef(), useRef()];
-  const [focusIndex, setFocusIndex] = useState(0);
+  const controlRefs: React.RefObject<HTMLButtonElement>[] = useMemo(
+    () => [
+      React.createRef<HTMLButtonElement>(),
+      React.createRef<HTMLButtonElement>(),
+      React.createRef<HTMLButtonElement>(),
+      React.createRef<HTMLButtonElement>(),
+    ],
+    []
+  );
+  const [focusIndex, setFocusIndex] = useState<number>(0);
```

Comment:
this is because it was `<number | null>` before. I stopped it from being a null value now so removed the `null` from this type here, not realising the default value is a type number anyway. Good spot.